src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentMethodsScreen.tsx

```diff
@@ -61,45 +61,29 @@ export const Profile_PaymentMethodsScreen = () => {
     authorizeRecurringPaymentError ||
     cancelRecurringPaymentError;
 
-  useEffect(() => {
-    const addPaymentMethodCallbackHandler = async ({url}: {url: string}) => {
-      if (
-        url.includes('response_code') &&
-        url.includes('recurring_payment_id')
-      ) {
-        InAppBrowser.close();
-        const responseCode = queryString.parseUrl(url).query.response_code;
-        const paymentId = Number(
-          queryString.parseUrl(url).query.recurring_payment_id,
-        );
-        if (responseCode === 'OK') {
-          await authorizeRecurringPayment(paymentId);
-        } else if (responseCode === 'Cancel') {
-          await cancelRecurringPayment(paymentId);
-        }
+  const addPaymentMethodCallbackHandler = (url: string) => {
+    if (url.includes('response_code') && url.includes('recurring_payment_id')) {
+      const responseCode = queryString.parseUrl(url).query.response_code;
+      const paymentId = Number(
+        queryString.parseUrl(url).query.recurring_payment_id,
+      );
+      if (responseCode === 'OK') {
+        authorizeRecurringPayment(paymentId);
+      } else if (responseCode === 'Cancel') {
+        cancelRecurringPayment(paymentId);
       }
-    };
+    }
+  };
 
-    // Listen to deep link redirects back here after
-    // the add card process completes.
-    const eventSubscription = Linking.addEventListener(
-      'url',
+  const onAddRecurringPayment = async () => {
+    const callbackUrl = `${APP_SCHEME}://payment-method-callback`;
```

Comment:
should this be "://profile" to match the one in `RootStack.tsx`? 🤔

particularly this line on `RootStack.tsx` : 

```
     TabNav_ProfileStack: 'profile',
``` 

Although based on the new flow, I don't think it matters. Just thinking out loud here.

Reply:
I found it a bit strange to use the `://profile` link when it doesn't link to the profile root screen. But then again, they both work, so maybe no point in changing it in this PR. 

I'll revert the change 👍

Reply:
@gorandalum Made the argument that it's better to have an url that can only trigger one thing, so there's no confusion where it is handled.

Reverted the revert 😅